src/tests/multihost/ipa/test_adtrust.py

```diff
@@ -115,3 +116,123 @@ def test_honour_idoverride(self, multihost, create_aduser_group):
+        :expectedresults:
+         0. Cache is cleared
+         1. The external group is member of posix group.
+         2. The posix group gid is present in id output .
+        :teardown:
+         Remove the created user, groups and revert resolution order.
+        :customerscenario: True
+        :bugzilla:
+         https://bugzilla.redhat.com/show_bug.cgi?id=1945552
+         https://bugzilla.redhat.com/show_bug.cgi?id=1937919
+         https://bugzilla.redhat.com/show_bug.cgi?id=1945654
+        """
+        ad_domain = multihost.ad[0].domainname
+        ipaserver = sssdTools(multihost.master[0])
+        ipa_domain = ipaserver.get_domain_section_name()
+        (username, _) = create_aduser_group
+        posix_group = "posix_group_01"
+        ext_group = "ext_group_01"
+        # SETUP
+        # Set the domain resolution order to AD first
+        resorder_cmd = f'ipa config-mod --domain-resolution-order=' \
+                       f'{ad_domain}:{ipa_domain}'
+        multihost.master[0].run_command(resorder_cmd, raiseonerr=False)
+
+        # Create posix group
+        pgroup_cmd = f'ipa group-add {posix_group}'
+        multihost.master[0].run_command(pgroup_cmd, raiseonerr=False)
+
+        # Create and external group
+        ext_group_cmd = f'ipa group-add --external {ext_group}'
+        multihost.master[0].run_command(ext_group_cmd, raiseonerr=False)
+
+        # Set membership of external group in posix group
+        member_cmd = f'ipa -n group-add-member {posix_group} --groups=' \
+                     f'{ext_group}'
+        multihost.master[0].run_command(member_cmd, raiseonerr=False)
+
+        # Set AD user membership in external group
+        usr_mbr_cmd = f"ipa -n group-add-member {ext_group} --external" \
+                      f" '{username}@{ad_domain}'"
+        multihost.master[0].run_command(usr_mbr_cmd, raiseonerr=False)
+
+        # TEST
+        # Get posix group id
+        grp_show_cmd = f"ipa group-show {posix_group}"
+        cmd = multihost.master[0].run_command(grp_show_cmd, raiseonerr=False)
+        gid_regex = re.compile(r"GID: (\d+)")
+        posix_group_id = gid_regex.search(cmd.stdout_text).group(1)
+
+        # Check that external group is member of posix group
+        grp_show_cmd = f"ipa group-show {ext_group}"
+        cmd = multihost.master[0].run_command(grp_show_cmd, raiseonerr=False)
+        assert posix_group in cmd.stdout_text, \
+            "The external group is not a member of posix group!"
+
+        # A bit of wait so the user is propagated
+        time.sleep(60)
+
+        # The reproduction rate is not 100%, I had reliably 2+
+        # fails in 5 rounds.
+        for _ in range(5):
+            # Clean caches on SSSD so we don't have to wait for cache timeouts
+            # The reproduction works better on sssd on ipa master
+            sssd_client = sssdTools(multihost.master[0])
+            sssd_client.clear_sssd_cache()
```

Comment:
In the description of the bug, the issue is not seen when the cache is cleared, but after a few minutes. Should a longer sleep be inside the loop?  and clear_sssd_cache be run only once outside the loop?

Reply:
I was explained that the issue is in negative cache. The getent group just before id somehow triggers that problem, so restarting sssd with removing caches make sure that there is a clean slate for getent to break it for the id.